compiler/src/iree/compiler/Codegen/Common/GPU/VectorReductionToGPU.cpp

```diff
@@ -276,12 +276,17 @@ class VectorReductionToGPUPass
         return AffineMap::get(vecRank, 0,
                               builder.getAffineDimExpr(vecRank - 1));
       };
+
       RewritePatternSet patterns(ctx);
       vector::populatePropagateWarpVectorDistributionPatterns(
           patterns, distributionFn, simpleWarpShuffleFunction);
       vector::populateDistributeReduction(patterns, groupReductionFn);
+
+      // We don't want to sink large transfer writes to a single lane -- pick a
+      // conservative value based on the group size.
+      unsigned maxWriteElementsToExtract = std::max(groupSize / 4, 1);
```

Comment:
Maybe silly question, but wouldn't we need this number to be bigger than the parallelFactor? if so, how can we guarantee that?

Reply:
Assuming we have a warpSize of 32, groupSize == warpSize, and maxWorkgroupSize of 1024. I think the only limit I see is `parallelFactor * groupSize < maxWorkgroupSize` which => parallelFactor < 32. 

However, this code limit caps the maxWriteElementToExtract to be 32/4 => maxWriteElementToExtract  would be 8.  technically possible to have a parallelFactor or parallel tilesize greater than the maxWriteElementToExtract. Which perhaps may be problematic?

Reply:
That's correct. These two are disconnected, and I didn't know if we could do better than having e2e tests for this. We could expose this as a constant / function here and use that to cap the `parallelFactor` on the config side. 

The way the math currently checks out is:
* `groupSize` is always 32 or 64 (in both files)
* `maxWorkgoupSize` is 1024
* `parallelFactor * groupSize < maxWorkgroupSize` ==> `parallelFactor < 1024 / groupSize` (either <= 16 or <= 8)
* `groupSize / 4` ==> (either 8 or 16)

So this is may not be enough for subgroup size 32. Nice catch, I need to fix this.

Reply:
Ah cool, thanks for the explanation Jakub :)

Reply:
hmm might be redundant but you can technically add the `groupSize / 4` constraint onto KernelConfig.

Reply:
I made both sides use the same formula (workgroup / 4) and added wave32 tests. I think this should be good is enough to make this consistent.